enlace/reports/attendance.php

```diff
@@ -250,6 +255,7 @@ function toggleAttendanceCheckboxes() {
         Participant_ID > 0) AND (Date_Dropped IS NULL OR Date_Dropped > 
         '$end_date_sqlsafe') " . $session_querystring;
     }
+    include "../include/dbconnopen.php";
```

Comment:
Just checking: is there any expectation that dbconnopen will be balanced by a corresponding dbconnclose?

Reply:
No, and there's a deeper thing to understand here.  dbconnopen is idemopotent.  dbconnclose is not.  So, the following happens a lot:
```php
open_db()
do_some_sql_stuff();
if(someConditional) {
  function_that_calls_open_then_close();
  // connection is closed here!
}
// Now the connection may be open or closed depending on updates to the sub function, and on how someConditional gets evaluated
do_some_more_sql_stuff(); // Likely to error someday
```

So, when I run into those errors, instead of trying to unwrap the codebase, I just throw another open in to make it work.

Reply:
Goodness, what a mess.  But fixing that mess is not part of this set of changes, agreed.